services/audit-service/src/main/java/datawave/microservice/audit/auditors/file/FileAuditor.java

```diff
@@ -0,0 +1,220 @@
+package datawave.microservice.audit.auditors.file;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import datawave.webservice.common.audit.AuditParameters;
+import datawave.webservice.common.audit.Auditor;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * An implementation for {@link Auditor}, which writes URL encoded, JSON formatted audit messages to a file.
+ */
+public class FileAuditor implements Auditor {
+    
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    
+    protected static final ObjectMapper mapper = new ObjectMapper();
+    
+    protected final SimpleDateFormat sdf;
+    protected final ReentrantLock writeLock = new ReentrantLock(true);
+    
+    protected long maxFileLenBytes;
+    protected long maxFileAgeMillis;
+    
+    protected FileSystem fileSystem;
+    protected Path basePath;
+    
+    protected Path currentFile = null;
+    protected Date creationDate = null;
+    
+    protected FileAuditor(Builder<?> builder) throws URISyntaxException, IOException {
+        this.maxFileLenBytes = builder.maxFileLenBytes;
+        this.maxFileAgeMillis = builder.maxFileAgeMillis;
+        
+        Configuration config = new Configuration();
+        config.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        config.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        
+        if (builder.configResources != null) {
+            for (String resource : builder.configResources)
+                config.addResource(new Path(resource));
+        }
+        
+        if (builder.fileUri != null)
+            fileSystem = FileSystem.get(new URI(builder.fileUri), config);
+        else
+            fileSystem = FileSystem.get(config);
+        
+        basePath = new Path(builder.path);
+        if (!fileSystem.exists(basePath))
+            fileSystem.mkdirs(basePath);
+        
+        this.sdf = new SimpleDateFormat("'" + builder.prefix + "'-yyyyMMdd_HHmmss.SSS'.json'");
```

Comment:
The sdf created here is only used in the createNewFile method.  In the current implementation, the createNewFile method is only ever called if the calling thread is able to acquire a lock on the writeLock object.  I think this should be enough to prevent multiple threads from accessing the sdf.